test/metadata-retrieval/metadataretrieval.go

```diff
@@ -0,0 +1,337 @@
+
+				if t > max {
+					max = t
+					maxi = i
+				}
+
+				results = append(results, t.String())
+			}
+		}
+
+		for i, r := range results {
+			fmt.Fprint(w, "\t")
+			if i == mini {
+				fmt.Fprintf(w, "\x1b[1;32m %s \x1b[0m", r)
+			} else if i == maxi {
+				fmt.Fprintf(w, "\x1b[1;31m %s \x1b[0m", r)
+			} else {
+				fmt.Fprintf(w, "\x1b[1;37m %s \x1b[0m", r)
+			}
+		}
+		fmt.Fprintf(w, "\n")
+	}
+	w.Flush()
+	fmt.Println()
+}
+
+// GetResults prints test results and returns if the tests passed
+func (t *Test) GetResults() bool {
+	if len(t.config.Versions) < 1 {
+		panic("there should be at least one version")
+	}
+
+	versions := t.config.Versions
+	ok := true
+	for i, version := range versions[0 : len(versions)-1] {
+		fmt.Printf("%s - %s ####\n", version, versions[i+1])
+		a := t.results[versions[i]]
+		b := t.results[versions[i+1]]
+
+		for _, orgs := range t.organizations {
+			fmt.Printf("## Organizations {%s} ##\n", orgs)
+			if _, found := a[orgs]; !found {
+				fmt.Printf("# Skip - organizations {%s} not found for version: %s\n", orgs, versions[i])
+				continue
+			}
+			if _, found := b[orgs]; !found {
+				fmt.Printf("# Skip - organizations {%s} not found for version: %s\n", orgs, versions[i+1])
+				continue
+			}
+
+			queryA := a[orgs][0]
+			queryB := b[orgs][0]
+
+			queryA.Result = average(a[orgs])
+			queryB.Result = average(b[orgs])
+			c := queryA.Result.ComparePrint(queryB.Result, 10.0)
+			if !c {
+				ok = false
+			}
+		}
+	}
+
+	return ok
+}
+
+// SaveLatestCSV saves test results in a CSV files
+// created file examples:
+//		- plot_org1_org2_org3_memory.csv
+//		- plot_org1_org2_org3_time.csv
+func (t *Test) SaveLatestCSV() {
+	version := t.config.Versions[len(t.config.Versions)-1]
+	for _, orgs := range t.organizations {
+		res := average(t.results[version][orgs])
+		if err := res.SaveAllCSV(fmt.Sprintf("plot_%s_", strings.Replace(orgs, ",", "_", -1))); err != nil {
+			panic(err)
+		}
+	}
+}
+
+// StoreLatestToPrometheus stores latest version results to prometheus pushgateway
+func (t *Test) StoreLatestToPrometheus(promConfig regression.PromConfig, ciConfig regression.CIConfig) error {
+	version := t.config.Versions[len(t.config.Versions)-1]
+	cli := prometheus.NewPromClient(Kind, promConfig)
+	for _, orgs := range t.organizations {
+		res := average(t.results[version][orgs])
+		if err := cli.Dump(res, version, orgs, ciConfig.Branch, ciConfig.Commit); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func average(pr []*Result) *regression.Result {
+	if len(pr) == 0 {
+		return nil
+	}
+
+	results := make([]*regression.Result, 0, len(pr))
+	for _, r := range pr {
+		results = append(results, r.Result)
+	}
+
+	return regression.Average(results)
+}
+
+func loadOrganizationsYaml(file string) ([]string, error) {
+	text, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []string
+	err = yaml.Unmarshal(text, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
```

Comment:
these methods are duplicated from `gitcollector` and `loadOrganizationsYaml` is not used here
can we probably move them to `test/test.go`?